suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,211 @@
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+    return currentRate?.isLoading ?? false;
+};
+
+export const selectIsTickerLoading = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+    rateType: RateType = 'current',
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectIsFiatRateLoading(state, fiatRateKey, rateType);
+};
+
+export const selectShouldUpdateFiatRate = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+
+    if (!currentRate) {
+        return true;
+    }
+
+    const { lastSuccessfulFetchTimestamp } = currentRate;
+    const now = Date.now();
+
+    return now - lastSuccessfulFetchTimestamp > MAX_AGE[rateType];
+};
+
+export const selectTickerFromAccounts = memoize((state: FiatRatesRootState): TickerId[] => {
+    const accounts = selectDeviceAccounts(state as any);
+    return pipe(
+        accounts,
+        A.map(account => [
+            {
+                symbol: account.symbol,
+            } as TickerId,
+            ...(account.tokens || []).map(
+                token =>
+                    ({
+                        symbol: account.symbol,
+                        tokenAddress: token.contract,
+                    }) as TickerId,
+            ),
+        ]),
+        A.flat,
+        F.toMutable,
+    );
+});
+
+export const selectTickersToBeUpdated = memoizeWithArgs(
+    (state: FiatRatesRootState, fiatCurrency: FiatCurrencyCode, rateType: RateType): TickerId[] => {
+        const tickers = selectTickerFromAccounts(state);
+        return tickers.filter(ticker => {
+            const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+            return (
+                selectShouldUpdateFiatRate(state, fiatRateKey, rateType) &&
+                !selectIsTickerLoading(state, ticker, fiatCurrency, rateType)
+            );
+        });
+    },
+    // used 5 just to be safe, but there shouldn't be more than one local currency at 2 rate types at one time
+    { size: 5 },
+);
+
+export const selectTransactionsWithMissingRates = memoizeWithArgs(
+    (
+        state: FiatRatesRootState & TransactionsRootState & AccountsRootState,
+        localCurrency: FiatCurrencyCode,
+    ) => {
+        const accountTransactions = selectTransactions(state);
+
+        return pipe(
+            accountTransactions,
+            D.mapWithKey((accountKey, txs) => ({
+                account: selectAccountByKey(state, accountKey as AccountKey),
+                txs: txs.filter(tx => !tx.rates?.[localCurrency]),
+            })),
+            D.filter(({ account, txs }) => !!account && !!txs.length),
+            D.values,
+            A.filter(value => !!value),
+        ) as {
```

Comment:
This cast can be removed by filtering with guard :point_up:

Reply:
See comment above ☝️

Reply:
I like to discuss this more, but for now it is not blocking the merge. Feel free to resolve if you really don't want to fix it now.